_posts/2018-09-13-API-Gateway-Types-Compared.md

```diff
@@ -0,0 +1,153 @@
+1. Connecting to a Regional API Gateway from the same region generated the lowest *minimal* response time, but in our test, Edge Optimized was still faster on average.
+
+All in all, we find the results quite surprising! 
+
+**Point 1**: We had not expected the Edge Optimized API Gateway to be so much more performant than the other two types. Based on this we would advise to use Edge Optimized over Regional in most use cases. If you are using Regional, do place your own CloudFront distribution in front of it.
+
+We can't say for sure *why* Edge Optimized is so much faster. In the end, traffic from the Edge locations will still need to travel to the API Gateway. It's unlikely that these paths are faster than other internal Amazon traffic, such as VPC Peerings. Other possible reasons for the fast Edge Optimized API are:
+* There is still some caching taking place, even though caching has been disabled.
+* Edge Optimized API Gateway has some internal smarts that allow certain elements to be executed on the Edge locations. We've executed our tests using Mock responses, and these could very well be delegated.
+
+But if either of these mechanisms are in place, we would expect the Edge Optimized tests to be equally fast from all different regions. And we're not seeing that either. In fact, Frankfurt is fastest, followed by Ireland, followed by North Virginia, followed by São Paulo, which confirms that distance plays a role.
+
+**Point 2**: Regional and Private gateways are in almost all situations equally fast. We did not expect this, because the traffic for the Private API Gateway stays within the Amazon networks, which we expected to be a bit faster. Of course, because we were using EC2 instances in our tests, it's well possible that the traffic for the Regional API Gateway *also* stayed within the Amazon network, which might explain this finding.
+
+**Point 3**: Only when testing the Private API Gateway over VPC Peering did we see failing connections (timeouts). We're not sure if VPC Peering or PrivateLink is to blame, but this might warrant further investigation. These timeouts are the cause of the very high Max values in the data.
+
+**Point 4**: Amazon's [documentation](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-basic-concept.html) indicates that for in-region communication, using a Regional API Gateway "bypasses the unnecessary round trip to a CloudFront distribution". Because of this, we expected to see significant improvements here, but our tests did not show them. However, accessing a Regional API Gateway from within the same Region did yield the lowest reponse time of all our tests at 8ms.
+
+# Final word
+We hope that you enjoyed this post, and that the difference in performance and use cases between the three types of API Gateway have become more clear. If you have any questions or remarks, give the author a shout on [Twitter](https://twitter.com/donkersgood). 
```

Comment:
Add or reach out to us on Slack (see my post)

Reply:
Oh cool, didn't know about that.